prow/sidecar/censor.go

```diff
@@ -71,35 +74,36 @@ func (o Options) censor() error {
 	if largest := censorer.LargestSecret(); 2*largest > bufferSize {
 		bufferSize = 2 * largest
 	}
+	censorFile := fileCensorer(sem, wg, errors, censorer, bufferSize)
+	censor := func(file string) {
+		wg.Add(1)
+		go censorFile(file)
```

Comment:
Yeah, I went back and forth on this, would love to hear your opinion:

1. I originally had the closure do the add as well, but we need access to the goroutine itself for the control flow management in the archival code
2. I went back and forth on bringing this out into a named function at all, since it's minimal, I just had N places where we did the exact same thing - wait group, semaphore, file handler - and knew that doing it wrong in any one case would be easy, so decided to DRY.

What do you think is the better option?

Reply:
I think the new approach of doing both the wg.Add and defer in `fileCensorer` is perfect